### DISCUSSION TITLE
Possibility to disable automatic conversion of units ?

### DISCUSSION THREAD

**Discussion by raphaeljolivet** [hgrecco/pint#1943] [General]
Is it possible to **disable automatic conversion** of units and fail when operation require changing the *scale* of the quantity ?

I am currently integrating this awesome library into [lca_algebraic](https://github.com/oie-mines-paristech/lca_algebraic) and would like the option to provide a "strict" mode, forcing users to convert quantities explicitely.

In practice, when this option is set, the following code **should fail** : 
```python
1 * u.ton + 1*u.kg
```

The user should do :
```
(1 * u.ton).to(u.kg) + 1*u.kg
```

Thanks in advance for your help.

**Comment by hgrecco** [comment]
There is no way to do that now, but it is an interesting idea and I do see the potential. You want to force the user to choose the units. Not sure if this can be done without a performance hit.

**Comment by raphaeljolivet** [comment]
Hi, 

I've just implemented it there : https://github.com/hgrecco/pint/compare/master...raphaeljolivet:pint:feature/strict_mode

I added an **auto_scale** argument to the **UnitRegistry** (True by default)

In the **_add_sub** method, I check it and raise an exception in case **auto_scale** is False and the scaling of the *other* argument is required. 

It is quite simple and suits my needs.
I have not implemented more complex cases when the **self** argument is scaled, and I am not sure to understand when this happens. 

I have not made a proper PR yet. 
I can do it if you find it useful. 
Let me know.

And thanks for this really great library !

Regards,
Raphael
